src/stlink-lib/usb.c

```diff
@@ -1134,33 +1134,8 @@ stlink_t *stlink_open_usb(enum ugly_loglevel verbose, enum connect_type connect,
 #endif
 
     libusb_device **list = NULL;
-    // TODO: We should use ssize_t and use it as a counter if > 0.
```

Comment:
In Linux user space, you can set `export STLINK_DEVICE="..."` before run command

Reply:
Yes, that does indeed make sense.

Reply:
@Ant-ON, maybe who writed this code some time ago knows it. But there is no such export anywhere in source files of  project. And there is comment with TODO, why to do so is bad idea (use serial instead of env. variable). So I think it can be done, but should not.

Reply:
@hydroconstructor You can safely assume that nobody is aware of it (anymore), and if so, very likely no longer available to be asked. I also believe nobody is willing to crawl the whole long project history to find out about it...

Reply:
@hydroconstructor I think this is not what @Ant-ON meant with his comment.
Nevertheless I'll leave a more detailed explanation to him.

Reply:
Wouldn’t dropping support for this environment variable kind of break backward compatibility? It is a feature people _may_ currently use.

Reply:
We will likely break backwards compatibility frequently as long as we don't clean-up the codebase from duplicate and what I'd call nested implementations and as long as ABI compatibility is not defined in any form (see #1075).
To achieve the latter one needs a well structured and modular code implementation and design.
So that's why there is no value to it from my perspective at the current point in time, as it would block any further attempts to clean up the codebase. Also the library is not documented well enough yet to serve as a core component for 3rd-party software.

Reply:
@gszy @hydroconstructor Please proceed with this.

Reply:
Other than the comment above, feel free to remove this code.